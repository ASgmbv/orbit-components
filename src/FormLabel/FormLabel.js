// @flow
import React from "react";
import styled from "styled-components";

const FormLabel = styled(({ className, children }) => (
  <span className={className}>{children}</span>
))`
  display: block;
  font-family: ${({ theme }) => theme.fontFamily};
  font-size: ${({ theme }) => theme.fontSizeLabelForm};
  color: ${({ theme, filled, disabled }) =>
    !filled || disabled ? theme.colorLabelForm : theme.colorLabelFormFilled};
  line-height: ${({ theme }) => theme.lineHeightText};
  margin-bottom: ${({ theme }) => theme.spaceXXSmall};
`;

export default FormLabel;
