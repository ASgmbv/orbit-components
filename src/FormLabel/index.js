// @flow
import React from "react";
import styled from "styled-components";

import defaultTheme from "../defaultTheme";
import AlertCircle from "../icons/AlertCircle";
import InformationCircle from "../icons/InformationCircle";

const StyledAsterisk = styled.span`
  font-weight: ${({ theme }) => theme.orbit.fontWeightBold};
  color: ${({ theme, filled }) =>
    !filled ? theme.orbit.colorTextError : theme.orbit.colorFormLabelFilled};
  font-size: ${({ theme }) => theme.orbit.fontSizeFormLabel};
  vertical-align: top;
`;

StyledAsterisk.defaultProps = {
  theme: defaultTheme,
};

const StyledInputErrorIcWrapper = styled.span`
  margin-right: ${({ theme }) => theme.orbit.spaceXXSmall};
  display: inline-flex;
`;

StyledInputErrorIcWrapper.defaultProps = {
  theme: defaultTheme,
};

const FormLabel = styled(
  ({
    className,
    children,
    required,
    filled,
    dataTest,
    id,
    error,
    help,
    onMouseEnter,
    onMouseLeave,
    domRef,
  }) => (
    <span ref={domRef} className={className} data-test={dataTest} id={id}>
      {error && (
        <StyledInputErrorIcWrapper onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
          <AlertCircle color="critical" size="small" />
        </StyledInputErrorIcWrapper>
      )}
      {!error && help && (
        <StyledInputErrorIcWrapper onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
          <InformationCircle color="secondary" size="small" />
        </StyledInputErrorIcWrapper>
      )}
      {required && (
        <StyledAsterisk aria-hidden="true" filled={filled}>
          *{" "}
        </StyledAsterisk>
      )}
      <span>{children}</span>
    </span>
  ),
)`
  display: flex;
  font-family: ${({ theme }) => theme.orbit.fontFamily};
  font-size: ${({ theme }) => theme.orbit.fontSizeFormLabel};
  color: ${({ theme, filled, disabled }) =>
    !filled || disabled ? theme.orbit.colorFormLabel : theme.orbit.colorFormLabelFilled};
  line-height: ${({ theme }) => theme.orbit.lineHeightText};
  margin-bottom: ${({ theme }) => theme.orbit.spaceXXSmall};
`;

FormLabel.defaultProps = {
  theme: defaultTheme,
};

export default FormLabel;
