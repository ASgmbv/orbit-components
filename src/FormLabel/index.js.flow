// @flow

export type Props = {
  children: React$Node,
  filled?: boolean,
  disabled?: boolean,
  required?: boolean,
};

declare export default React$ComponentType<Props>;
