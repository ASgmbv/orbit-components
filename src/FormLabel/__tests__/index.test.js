// @flow
import * as React from "react";
import { shallow } from "enzyme";

import FormLabel from "../FormLabel";

describe("FormLabel", () => {
  const component = shallow(<FormLabel>FormLabel</FormLabel>);

  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
