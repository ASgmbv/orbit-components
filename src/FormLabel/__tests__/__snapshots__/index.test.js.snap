// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel should match snapshot 1`] = `
<Component
  className="FormLabel-kJFBRQ bTqPNB"
>
  FormLabel
</Component>
`;
